vhdbuilder/packer/backfill-cleanup.sh

```diff
@@ -0,0 +1,104 @@
+if [[ -n "${AZURE_RESOURCE_GROUP_NAME}" && "${CLEAN_OLD_WINDOWS_RESOURCES}" == "True" && "${SIG_GALLERY_NAME}" != "AKSWindows" ]]; then
+  echo "Looking for Windows managed images in ${AZURE_RESOURCE_GROUP_NAME} created over ${EXPIRATION_IN_HOURS} hours ago..."
+
+  managed_image_ids=""
+  sig_version_ids=""
+
+  # delete outdated Windows managed images and associated SIG versions (.tags.os must be "Windows")
+  images=$(az image list -g ${AZURE_RESOURCE_GROUP_NAME} | jq --arg dl $deadline '.[] | select(.tags.os == "Windows") | select(.tags.now < $dl).name' | tr -d '\"' || "")
+  for image in $images; do
+    managed_image_ids="${managed_image_ids} /subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${AZURE_RESOURCE_GROUP_NAME}/providers/Microsoft.Compute/images/${image}"
+    sig_version_ids="${sig_version_ids} /subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${AZURE_RESOURCE_GROUP_NAME}/providers/Microsoft.Compute/galleries/${SIG_GALLERY_NAME}/images/${image%-*}/versions/${image#*-}"
+    echo "Will delete Windows managed image ${image} and associated SIG version from resource group ${AZURE_RESOURCE_GROUP_NAME}"
+  done
+
+  if [[ -n "${managed_image_ids}" ]]; then
+    echo "Attempting to delete $(echo ${managed_image_ids} | wc -w) Windows managed images..."
+    az resource delete --ids ${managed_image_ids} > /dev/null || echo "Windows managed image deletion was not successful, continuing..."
+  else
+    echo "Did not find any Windows managed images eligible for deletion"
+  fi
+
+  if [[ -n "${sig_version_ids}" ]]; then
+    echo "Attempting to delete $(echo ${sig_version_ids} | wc -w) Windows SIG image versions associated with old managed images..."
+    az resource delete --ids ${sig_version_ids} > /dev/null || echo "Windows SIG image version deletion was not successful, continuing..."
+  else
+    echo "Did not find any SIG versions associated with old Windows managed images eligible for deletion"
+  fi
+fi
+
+# attempt to clean up Windows SIG image versions in all galleries except "AKSWindows" created over a week ago
+if [[ -n "${AZURE_RESOURCE_GROUP_NAME}" && "${CLEAN_OLD_WINDOWS_RESOURCES}" == "True" ]]; then
+  gallery_list=$(az sig list -g ${AZURE_RESOURCE_GROUP_NAME} | jq '.[] | select(.name != "AKSWindows") | .name' | tr -d '\"' || "")
+  for gallery in $gallery_list; do
+    # delete old Windows SIG image versions in gallery (image definitions must have .osType == "Windows")
+    image_defs=$(az sig image-definition list -g ${AZURE_RESOURCE_GROUP_NAME} -r ${gallery} | jq '.[] | select(.osType == "Windows").name' | tr -d '\"' || "")
+    for image_definition in $image_defs; do
+        echo "Finding sig image versions associated with ${image_definition} in gallery ${gallery}"
+        image_versions=$(az sig image-version list -g ${AZURE_RESOURCE_GROUP_NAME} -r ${gallery} -i ${image_definition} | jq --arg dl $deadline '.[] | select(.tags.now < $dl).name' | tr -d '\"' || "")
```

Comment:
nit: everywhere we do a selection from an array with jq, you can tack `-r` onto it so we don't need the `tr -d '\"'`